packages/arb-token-bridge-ui/tests/e2e/specs/approveToken.cy.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  importTokenThroughUI,
+  ERC20TokenName,
+  ERC20TokenSymbol,
+  zeroToLessThanOneETH
+} from '../../support/common'
+
+const ERC20TokenAddressL1 = Cypress.env('ERC20_TOKEN_ADDRESS_L1')
+
+describe('Approve token', () => {
+  // log in to metamask
+  before(() => {
+    cy.login({ networkType: 'L1' })
+  })
+  after(() => {
+    // after all assertions are executed, logout and reset the account
+    cy.logout()
+  })
+  beforeEach(() => {
+    cy.restoreAppState()
+  })
+  afterEach(() => {
+    cy.saveAppState()
+  })
+
+  context('User approves and deposits ERC-20 token', () => {
+    it('should import token through UI', () => {
+      importTokenThroughUI(ERC20TokenAddressL1)
+
+      // Select the ERC-20 token
+      cy.findByText('Added by User').should('exist')
+      cy.findByText(ERC20TokenName).click({ scrollBehavior: false })
+
+      // ERC-20 token should be selected now and popup should be closed after selection
+      cy.findByRole('button', { name: 'Select Token' })
+        .should('be.visible')
+        .should('have.text', ERC20TokenSymbol)
+    })
+
+    it('should approve successfully', () => {
+      cy.findByPlaceholderText('Enter amount')
+        .type('0.000000000001', { scrollBehavior: false })
+        .then(() => {
+          cy.findByText('You’ll pay in gas fees')
+            .siblings()
+            .last()
+            .contains(zeroToLessThanOneETH)
+            .should('be.visible')
+          cy.findByText('L1 gas')
+            .parent()
+            .siblings()
+            .last()
+            .contains(zeroToLessThanOneETH)
+            .should('be.visible')
+          cy.findByText('L2 gas')
+            .parent()
+            .siblings()
+            .last()
+            .contains(zeroToLessThanOneETH)
+            .should('be.visible')
+        })
+      cy.findByRole('button', {
+        name: 'Move funds to Arbitrum'
+      })
+        .as('transferButton')
+        .click({ scrollBehavior: false })
+        .then(() => {
+          cy.findByText(/I understand that I have to pay a one-time/).click({
+            scrollBehavior: false
+          })
+          cy.findByRole('button', {
+            name: /Pay approval fee of/
+          }).click({ scrollBehavior: false })
+          cy.confirmMetamaskPermissionToSpend()
+        })
+      cy.get('@transferButton').should('be.disabled')
+    })
+
+    it('should deposit successfully', () => {
+      // TODO: we don't have any indication in the UI that we are approving a token.
```

Comment:
It would be nice to discuss UX solutions for this. Currently we don't change the UI state based on type of tx's we're processing. E.g. when initiating approval nothing will change in the UI throughout the entire ERC-20 deposit.

Reply:
Should we start a GitHub issue for this?